hpat/tests/test_series.py

```diff
@@ -186,14 +186,23 @@ def test_impl(A):
         S = pd.Series([3, 5, 6], ['a', 'b', 'c'], name='A')
         pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
 
-    def test_series_attr1(self):
-        def test_impl(A):
-            return A.size
+    def test_series_size(self):
+        def test_impl(S):
+            return S.size
         hpat_func = hpat.jit(test_impl)
 
         n = 11
-        df = pd.DataFrame({'A': np.arange(n)})
-        self.assertEqual(hpat_func(df.A), test_impl(df.A))
+        S = pd.Series(np.arange(n))
+        self.assertEqual(hpat_func(S), test_impl(S))
+
+        S = pd.Series()
+        self.assertEqual(hpat_func(S), test_impl(S))
+
+        S = pd.Series([np.nan, 1, 2])
+        self.assertEqual(hpat_func(S), test_impl(S))
+
+        S = pd.Series(['1', '2', '3'])
```

Comment:
@PokhodenkoSA It might be better to make these set of tests as a loop.
I mean:
```
for input_data in test_input_data:
    S = pd.Series(input_data)
```
where data defined as https://github.com/IntelPython/hpat/blob/e45cc922afb855f0e5d87e30acc01ef6ef2e4002/hpat/tests/test_series.py#L2057

also, it might be better to use `pd.testing.assert_series_equal` instead `self.assertEqual` because, as I understood, the first one compare full `Series` datatype but second compares part of it. Yes, we use diffrent types of assertion in tests but it will be a one of the task for redeveloping.

Reply:
I think `pd.testing.assert_series_equal` is not applicable here because it compares who Series but Series.size returns scalar.